src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.Easy.cs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Http
+    {
+        [DllImport(Libraries.HttpNative)]
+        public static extern SafeCurlHandle EasyCreate();
+
+        [DllImport(Libraries.HttpNative)]
+        private static extern void EasyDestroy(IntPtr handle);
+
+        [DllImport(Libraries.HttpNative, CharSet = CharSet.Ansi)]
+        public static extern CURLcode EasySetOptionString(SafeCurlHandle curl, CURLoption option, string value);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasySetOptionLong(SafeCurlHandle curl, CURLoption option, long value);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasySetOptionPointer(SafeCurlHandle curl, CURLoption option, IntPtr value);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasySetOptionPointer(SafeCurlHandle curl, CURLoption option, SafeHandle value);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasySetOptionPointer(SafeCurlHandle curl, CURLoption option, Delegate callback);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern IntPtr EasyGetErrorString(int code);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasyGetInfoPointer(IntPtr handle, CURLINFO info, out IntPtr value);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasyGetInfoLong(SafeCurlHandle handle, CURLINFO info, out long value);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasyPerform(SafeCurlHandle curl);
+
+        [DllImport(Libraries.HttpNative)]
+        public static extern CURLcode EasyUnpause(SafeCurlHandle easy);
+
+        // Curl options are of the format <type base> + <n>
+        private const int CurlOptionLongBase = 0;
+        private const int CurlOptionObjectPointBase = 10000;
+        private const int CurlOptionFunctionPointBase = 20000;
+
+        // Enum for constants defined for the enum CURLoption in curl.h
+        internal enum CURLoption
+        {
+            CURLOPT_INFILESIZE = CurlOptionLongBase + 14,
```

Comment:
One naïve question: is there an advantage to not using the macros from curl.h  directly? Does that involve a cast which we are avoiding with this approach?